apis/v1beta1/gateway_types.go

```diff
@@ -148,6 +154,69 @@ type GatewaySpec struct {
 	Addresses []GatewayAddress `json:"addresses,omitempty"`
 }
 
+// GatewayInfrastructure defines infrastructure level attributes about a Gateway
+type GatewayInfrastructure struct {
+	// Routability allows the Gateway to specify the accessibility of its addresses. Setting
+	// this property will override the default value defined by the GatewayClass.
+	//
+	// If the desired Gateway routability is incompatible with the GatewayClass implementations
+	// MUST set the condition `Accepted` to `False` with `Reason` set to `UnsupportedRoutability`.
+	//
+	// The default value of routability is implementation specific and MUST remain consistent for
+	// Gateways with the same gatewayClassName
+	//
+	// Implementations MUST clearly document if they support updates to this field. The default
+	// expectation should be that changes to this field are not supported unless an implementation
+	// specifies otherwise.
+	//
+	// If a Gateway is mutated but does not support the desired routability it MUST set the `Accepted`
+	// and `Programmed` conditions to `False` with `Reason` set to `UnsupportedRoutability`.
+	//
+	// +optional
+	Routability *GatewayRoutability `json:"routability,omitempty"`
+}
+
+// GatewayRoutablility represents the routability of a Gateway
+//
+// The pre-defined values listed in this package can be compared semantically.
+// `Public` has a larger scope than `Private`, while `Private` has a larger scope than
+// `Cluster`.
```

Comment:
It's helpful for me to think of this as
- Public: External
- Private: Multi-cluster
- Cluster: Internal

Reply:
Private losely maps to some internal network (eg. VPN) - so multi-cluster isn't accurate.

See the discussion on the origin GEP PR - https://github.com/kubernetes-sigs/gateway-api/pull/1653